library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -39,7 +39,16 @@ object JSConverters extends JSConvertersLowPrioImplicits {
 
   implicit class JSRichGenTraversableOnce[T](
       val col: GenTraversableOnce[T]) extends AnyVal {
-    @inline final def toJSArray: Array[T] = genTraversableOnce2jsArray(col)
+    final def toJSArray: Array[T] = {
+      col match {
+        case col: js.ArrayOps[T]     => col.repr
+        case col: js.WrappedArray[T] => col.array
+        case _ =>
+          val result = new js.Array[T]
+          col.foreach(x => result.push(x))
+          result
+      }
+    }
```

Comment:
So just to check my understanding: This behaves better than before in case `col` is stack allocated?

Reply:
No. In case `col` is stack allocated, this performs *exactly* the same as before. The difference is when `col` is *not* stack allocated, in which case the new version produces less code than before by not inlining the method.

Reply:
Ah, right. Because `genTraversableOnce2jsArray` is inlineable as well. OK, got it.